File: 2037-count-square-sum-triples/count-square-sum-triples.cpp
class Solution {
public:
    int countTriples(int n) {
        map<int,int> mp;
        int cnt=0;
        for(int i=1;i<=n;i++)
        {
            mp[i*i]=1;
            for(int j=1;j<i;j++)
            {
                if(mp[(i*i)-(j*j)])cnt++;
            }
        }
        return cnt;
    }
};

File: 2037-count-square-sum-triples/count-square-sum-triples.java
class Solution {
    public int countTriples(int n) {
        HashMap<Integer, Integer> mp = new HashMap<>();
        int cnt=0;
        for(int i=1;i<=n;i++)
        {
            int a = i*i;
            for(int j=i+1;j<=n;j++)
            {
                int b = j*j;
                int c = (int)Math.sqrt(a+b);
                if(c*c == (a+b) && c<=n)
                {
                    cnt+=2;
                    //System.out.println(i+" "+j);
                }
            }
        }
        return cnt;
    }
}